src/util.h

```diff
@@ -186,6 +186,26 @@ static SECP256K1_INLINE void *manual_alloc(void** prealloc_ptr, size_t alloc_siz
 #if defined(_MSC_VER) && defined(_WIN32) && !defined(SECP256K1_LITTLE_ENDIAN)
 # define SECP256K1_LITTLE_ENDIAN
 #endif
+#if !defined(SECP256K1_LITTLE_ENDIAN) && !defined(SECP256K1_BIG_ENDIAN)
```

Comment:
Do we need this `#if` ? It breaks with the idea of the code above, which tries to detect from all available sources instead and then fail below if both are defined, e.g., if a user supplied `#define` is inconsistent with our detection. Seems a tiny bit safer.

Reply:
Hm? if the detection is wrong (e.g. because a system header sets something it shouldn't) the user should be able to override.

Reply:
For example if the user wrongly sets `SECP256K1_BIG_ENDIAN`, then `__x86_64__` will not be checked at all due to this line (if I'm not getting it wrong entirely).

Reply:
That is my understanding of the behaviour. If some dumb header has __x86_64__ in it why shouldn't the user should be able to override?

I've seen stuff set things like that just to make stuff compile when some code was erroneously testing __i386__  to determine if the processor was 32-bit.

Reply:
Ok, we can discuss this but it should be consistent at least. The existing code is different, see my commit message in #787.

> The macros are carefully written to err on the side of the caution,
e.g., we #error if the user manually configures a different endianness
than what we detect.

Reply:
What you're suggesting would override the BYTE_ORDER macro. I don't think it makes any sense at all to have arch guessing override explicitlyly set BYTE_ORDER or the user.  There is a difference between a conflicting setting and a guess conflicting with something explicit.

Reply:
So really we have 3 potentially different sources:
* User manually setting `SECP256K1_{LITTLE,BIG}_ENDIAN`
* `__BYTE_ORDER__`
* The stuff I'm adding in this PR

Should all 3 be consistent with each other, if they're available? That means no mechanism is left for the user to override if the detection is somehow wrong. Should we have a USE_FORCE_{BIG,LITTLE}_ENDIAN for that case?

One possibility is just using SECP256K1_{BIG,LITTLE}_ENDIAN. If exactly one of those two is set, that's what is used. If not, all other sources that are found have to agree.